users/views.py

```diff
@@ -1,3 +1,54 @@
-from django.shortcuts import render
+import json, re, uuid, random
+import bcrypt
 
-# Create your views here.
+from django.http      import JsonResponse
+from django.views     import View
+from django.db        import IntegrityError
+from django.db.models import Q
+
+from users.models   import User
+
+EMAIL_REGEX    = r'\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b'
+PASSWORD_REGEX = r'^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!#%*?&]{6,20}$'
+
+class EmailSignupView(View):
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+
+            if not re.match(EMAIL_REGEX, data['email']):
+                return JsonResponse({"message": "INVALID_EMAIL"}, status=400)
+            
+            if User.objects.filter(email=data['email']).exists():
+                return JsonResponse({"message": "DUPLICATE_EMAIL"}, status=400)
+
+            if not re.match(PASSWORD_REGEX, data['password']):
+                return JsonResponse({"message": "INVALID_PASSWORD"}, status=400)
+
+            hashed_password = bcrypt.hashpw(data['password'].encode('utf-8'), bcrypt.gensalt())
+
+            User.objects.create(
+                email              = data['email'],
+                name               = data['name'],
+                password           = hashed_password.decode(),
+                deposit_bank_id    = random.randint(1, 8),
+                deposit_account    = create_random_account(),
+                withdrawal_bank_id = random.randint(1, 8),
+                withdrawal_account = create_random_account()
+            )
+
+            return JsonResponse({"message": "SUCCESS"}, status=201)
+
+        except KeyError:
+            return JsonResponse({"message": "KEY_ERROR"}, status=400)
+        
+        except IntegrityError:
+            return JsonResponse({"message": "DUPLICATE"}, status=400)
+
+def create_random_account():
+    random_account = str(uuid.uuid4().int>>64)[0:17]
```

Comment:
random account를 복잡하게 만드는 이유가 있나요?
```suggestion
    random_account = str(uuid.uuid4().int>>64)[0:17]
```

Reply:
uuid가 그나마 중복값을 걸러줄거라 생각해서 uuid를 사용하긴 했는데, random.randint(1e16,1e17)가 나을까요?

Reply:
아래 속성을 사용하면 바로 hex 값을 가져올 수 있습니다~
```suggestion
    uuid.uuid4().hex
```